torchio/transforms/augmentation/intensity/random_simulation.py

```diff
@@ -0,0 +1,172 @@
+
+from typing import Union, Tuple, Optional, Dict, List
+import torch
+from ....torchio import DATA, TypeData, TypeRangeFloat, TypeNumber, AFFINE, INTENSITY, TYPE, LABEL
+from ....data.subject import Subject
+from ....data.image import Image
+from .. import RandomTransform
+
+MEAN_RANGE = (0.1, 0.9)
+STD_RANGE = (0.01, 0.1)
+
+
+class RandomSimulation(RandomTransform):
+    r"""Generate a random image from label maps.
+
+    Args:
+        label_keys: Keys designating the label images in the sample that
+            will be used to generate the new image.
+            If a single string is given, the corresponding label map is expected
+            to be a label encoding map.
```

Comment:
I think this is this is to differentiate from single mask: label are encode with unique int volues into a single volume

Reply:
Two common ways exist to create categorical encoding: label encoding or one-hot encoding. Label encoding generally gives different integer values to the different categories.

For instance, `['Paris', 'London', 'New York']` could be encoded into `[0, 1, 2]`.

Maybe the word "map" is a bit confusing and should be replaced with "volume"?

Reply:
Oh, I think it's what I call simply a "label map". I.e. a volume with discrete labels such as `[0 0 1 1 2 1 0 0]`. Why do you allow to generate the image from different label maps? It makes the transform a bit too complex, I think.

Reply:
I guess the answer is right below.

Reply:
The partial volume version of a label map, is a 4D nifti volume, where the Partial volume labels are in the forth dimension. 
We already suggest to try to incorporate this kind of data, and I think it will be more consistent, but it can be replace with n_label_keys ... so if we have such volume as input we will split them in different keys (indexed by the label level, (ie the index of the 4th dimention)
This may not be optimal with n being big ... When (/ if) we have 4D data internally supported this will be allow as inputs